File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
   vector<vector<int>> merge(vector<vector<int>> &arr)
{   
    sort(arr.begin(),arr.end());
    vector<vector<int>> ans;
    for(int i=0;i<arr.size();i++)
    {
        if(!ans.empty() && arr[i][0]<=ans.back()[1])
        {
            ans.back()[1]=max(ans.back()[1],arr[i][1]);
        }
        else{
            ans.push_back(arr[i]);
        }
    }
    return ans;
   
    

}
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {

    public int[][] merge(int[][] intervals) {
        int n = intervals.length;
        List<int[]> ans = new ArrayList<>();
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        ans.add(intervals[0]);
        for (int i = 1; i < n; i++) {
            int[] prev = ans.getLast();
            int[] curr = intervals[i];
            if (prev[1] >= curr[0]) {
                ans.removeLast();
                int[] temp = new int[] { prev[0], Math.max(prev[1], curr[1]) };
                ans.add(temp);
            } else {
                ans.add(intervals[i]);
            }
        }
        int[][] result = new int[ans.size()][2];
        for (int i = 0; i < ans.size(); i++) {
            result[i] = ans.get(i);
        }
        return result;

    }
}